doc/development.md

```diff
@@ -89,4 +89,205 @@ merged, we need to test them on all supported distributions and
 architectures.
 
 When you push patches to GitHub, CI will run its tests according to the
-configuration in the `.github/workflows/ci.yml` file.
\ No newline at end of file
+configuration in the `.github/workflows/ci.yml` file.
+
+## Coding guidelines
+
+The coding conventions in Vdsm are generally aligned with the
+[PEP 8](https://peps.python.org/pep-0008/) standard guidelines.
+Code is linted with `pylint` based in the `/.pylintrc` file.
+You can lint the code before submitting a patch by doing:
+
+    make lint
+
+However, some code conventions are not explicitly enforced by the linter due
+to legacy code and inconsistent styles. But new code shall adhere to those
+rules, and old code, should be updated as possible when patched.
+
+In this section, we will cover some of the most important conventions and
+preferences for contributing to Vdsm.
+
+### Imports
+
+First import standard library modules, then 3rd party modules, and finally
+project modules.
+Separate standard library modules and 3rd party modules with a blank line.
+Keep alphabetical order within each block.
+
+Less specific imports:
+
+    from vdsm.foo import baz
+
+Should come before more specific imports like:
+
+    from vdsm.foo.bar import baz
+
+Accepted, but better to avoid:
+- Relative name imports
+- Import many modules in a comma separated list
+- Alias an imported module with non-standard abbreviation
+
+### Indentation
+
+Use 4 spaces per indentation level.
+
+Continuation lines should align wrapped elements either vertically using
+Python's implicit line joining inside parentheses, brackets and braces, or
+using a hanging indent. When using a hanging indent, the following should
+be considered:
+- There should be no arguments on the first line.
+- Further indentation should be used to clearly distinguish itself as a continuation line.
+- Hanging indentation should stick to the 4-space rule.
+```python
+# Correct
+# Add 4 spaces (an extra level of indentation) to distinguish arguments from the rest.
+foo = long_function_name(
+        var_one, var_two, var_three, var_four)
+
+# Also with if-statements.
+if (condition_1 and
+        condition_2):
+    do_something()
+
+# One line per argument for cases where arguments expand to more than one line.
+def long_function_name(
+        var_one,
+        var_two,
+        var_three,
+        var_four):
+    print(var_one)
```

Comment:
We don't use this style. If a function definition does not fit in one line we do this:

```
    def __init__(self, domain, image, parent, capacity, format, type, voltype,
                 disktype, description="", legality=sc.ILLEGAL_VOL, ctime=None,
                 generation=sc.DEFAULT_GENERATION,
                 sequence=sc.DEFAULT_SEQUENCE):
```

But in general we don't have functions with so many arguments and it an anti-pattern to add such functions. They are hard to work with regardless of how you indent them. The coding style guide should discourage adding lot of arguments.

Reply:
I'd say I added this style from a review, but ok, I'll remove it. I agree with many arguments as an antipattern, and I'll add the comment. But it can be that does not fit a single line due to long variable names and default values being used, so it is still interesting to comment how to properly break the line.